includes/specials/SpecialUndelete.php

```diff
@@ -328,9 +328,9 @@ function isDeleted() {
 	function undelete( $timestamps, $comment = '', $fileVersions = array(), $unsuppress = false ) {
 		global $wgUser;
 
-		$resultMock = '';
-		if ( !$this->title->userCan( 'undelete' ) ) {
```

Comment:
I'm with Kamil on this one. This is a very strange and bad UX not to give users a reason for which you blocked an action for them.